### PR TITLE
Add in new p2p data models and adjust channel model

### PR DESCRIPTION
**Pull request by jlbyrne**
Resolves https://github.com/brave-intl/creators-private-issues/issues/140

### CODE REVIEW

File: app/models/channel.rb [hunk 1]
```diff
@@ -57,10 +58,15 @@ class Channel < ApplicationRecord
     in: %w[domain_not_found connection_failed too_many_redirects timeout no_txt_records token_incorrect_dns token_not_found_dns token_not_found_public_file no_https]
   }, allow_nil: true
 
+  validates :public_name, uniqueness: true, allow_nil: true
+  # validates :public_identifier, uniqueness: true, presence: true
```
Comment (tsmartt) [inline app/models/channel.rb:62]: Comment

Comment (tsmartt) [reply app/models/channel.rb:62]: In ca you were wanting to uncomment this for uniqueness, nevermind. But you can also add an index in the migration to enforce uniqueness.

File: db/migrate/20230320213156_create_crypto_address_for_channels.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class CreateCryptoAddressForChannels < ActiveRecord::Migration[7.0]
+  def change
+    create_table :crypto_address_for_channels, id: :uuid, default: -> { "uuid_generate_v4()"} do |t|
+      t.string :chain, null: false
```
Comment (tsmartt) [inline db/migrate/20230320213156_create_crypto_address_for_channels.rb:4]: We should consider an enum here https://github.com/bibendi/activerecord-postgres_enum

to enforce data integrity outside of the app

File: test/models/channel_test.rb [hunk 3]
```diff
@@ -94,6 +94,29 @@ class ChannelTest < ActionDispatch::IntegrationTest
...
+    channel.save!
+
+    channel_2 = channels(:new_site)
+    channel_2.public_name = channel.public_name
+    refute channel_2.valid?
+
+    assert_equal "has already been taken", channel_2.errors.messages[:public_name][0]
+  end
+
+  # test "public_identifier is added on create and must be unique" do
```
Comment (tsmartt) [inline test/models/channel_test.rb:109]: Comments

**Review by tsmartt — COMMENTED** [review]
Looks pretty good, left a few comments.

File: app/models/channel.rb (hunk 1 above)
Comment (jlbyrne) [reply app/models/channel.rb:62]: This validation will work fine once I run the rake task to backfill values.  Until I do that, though, the validation will cause a bunch of errors.  I'll uncomment it in a future PR once the task has been run.

File: test/models/channel_test.rb (hunk 3 above)
Comment (jlbyrne) [reply test/models/channel_test.rb:109]: Same issue as above- once the rake task has been run, I can uncomment the validation and the test will pass.

File: db/migrate/20230320213156_create_crypto_address_for_channels.rb (hunk 2 above)
Comment (jlbyrne) [reply db/migrate/20230320213156_create_crypto_address_for_channels.rb:4]: that's a good idea, I'll make some adjustments.

**Review by porteron — APPROVED** [review]

**Review by porteron — APPROVED** [review]

**Pull request merged**